plugins/serializers/splunkmetric/splunkmetric.go

```diff
@@ -0,0 +1,125 @@
+package splunkmetric
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/influxdata/telegraf"
+)
+
+type serializer struct {
+	HecRouting bool
+}
+
+func NewSerializer(hec_routing bool) (*serializer, error) {
+	s := &serializer{
+		HecRouting: hec_routing,
+	}
+	return s, nil
+}
+
+func (s *serializer) Serialize(metric telegraf.Metric) ([]byte, error) {
+
+	m, err := s.createObject(metric)
+	if err != nil {
+		log.Printf("D! [serializer.splunkmetric] Dropping invalid metric: %v [%v]", metric, m)
```

Comment:
We should return this as an error: `return nil, fmt.Errorf("Dropping invalid metric: %s", metric.Name())`